Remove code generating receipts

anansilva opened the pull request:
#### What? Why?

- Closes [#9311](https://github.com/openfoodfoundation/openfoodnetwork/issues/9311)

This PR: 

- Removes code that generates tickets/receipts 
- Removes the receipts settings

#### What should we test?

1. Visit Configurations and choose Invoice Settings from the sidebar menu 

- The invoice settings should no longer include the option that activates the printing of receipts

![Captura de ecrã 2023-02-10, às 17 22 30](https://user-images.githubusercontent.com/27692541/218250326-0f17c7ec-00d2-4eac-8e8a-e954545adf9c.png)

2. Visit the order edit page and click on the 'Actions Button'

- The links to print receipts should not be shown

![Captura de ecrã 2023-02-10, às 17 25 10](https://user-images.githubusercontent.com/27692541/218250394-a100a0a7-70b9-4d49-b91f-596bc006d8a5.png)

#### Release notes

Changelog Category: User facing changes

anansilva commented on config/locales/en.yml:
```diff
@@ -590,7 +590,6 @@ en:
         title: "Invoice Settings"
         enable_invoices?: "Enable Invoices?"
         invoice_style2?: "Use the alternative invoice model that includes total tax breakdown per rate and tax rate info per item (not yet suitable for countries displaying prices excluding tax)"
-        enable_receipt_printing?: "Show options for printing receipts using thermal printers in order dropdown?"
```
Is it enough to remove the translations from the `en` file? Will the current worklow with transifex remove these from other locales? Thanks!

mkllnk reviewed (commented):
Amazing work, thank you.

I think that we can remove the qz gem dependency as well (and the reference in config/application.rb).

jibees reviewed (commented):
Very nice PR, very clear with atomic commits: 🙏 💪 Thanks a lot!

Thoughts:
 - We do have a discussion around removing the feature here: https://github.com/openfoodfoundation/openfoodnetwork/issues/9311#issuecomment-1427166989 that needs to be resolved before going further on it,
 - I think having a data migration to remove that preference should be implemented
 - We probably should rebase the PR just to remove that flaky spec, and have a green build
 - What about removing `shared/jquery-1.8.0.js` as well? It seems to not be used elsewhere right?

What do you think?

anansilva commented on app/models/spree/app_configuration.rb:
```diff
@@ -123,10 +123,9 @@ class AppConfiguration < Preferences::Configuration
+    # Invoices
     preference :enable_invoices?, :boolean, default: true
     preference :invoice_style2?, :boolean, default: false
-    preference :enable_receipt_printing?, :boolean, default: false
```
Good point, I've added a migration to delete it.

anansilva commented:
Thanks @mkllnk and @jibees. I've updated the PR with a migration to delete the receipt printing from the preferences. I've also removed `jquery-1.8.0.js`, that indeed does not look like it is being used elsewhere. There are some js tests failing though. Could it be because of this or are these known flaky tests? (I've rebased this PR today)

jibees commented:
> I've also removed `jquery-1.8.0.js`, that indeed does not look like it is being used elsewhere. There are some js tests failing though. Could it be because of this or are these known flaky tests? (I've rebased this PR today)

Yep, I think because jQuery is needed somewhere in the js spec files. I think we should add `jquery2` in file `spec/javascripts/application_spec.js`: 

```diff
diff --git a/spec/javascripts/application_spec.js b/spec/javascripts/application_spec.js
index 6f5831dace..33a5b2bdda 100644
--- a/spec/javascripts/application_spec.js
+++ b/spec/javascripts/application_spec.js
@@ -14,6 +14,7 @@
 //= require moment/min/moment.min.js
 //= require i18n
 //= require handlebars
+//= require jquery2
 
 angular.module('templates', [])
 angular.module('uiGmapgoogle-maps', []);
```

(like it's done in `app/assets/javascripts/admin/all.js`, used in the app)

anansilva commented:
Ah, good one @jibees! Thanks, all green now 💚

mkllnk commented on db/migrate/20230213160135_delete_receipt_printing_from_preferences.rb:
```diff
@@ -0,0 +1,5 @@
+class DeleteReceiptPrintingFromPreferences < ActiveRecord::Migration[6.1]
+  def change
```
```suggestion
  def up
```

Otherwise the delete will also happen on rollback.

mkllnk reviewed (changes requested):
Excellent work! Just one tiny change, please. (And resolving the merge conflict by rebasing).

jibees reviewed (approved):
I would love to see commit ebf53a22f212756d9ac3c41568ba44f8c985a5a7 squashed with commit a1c832a48b5fc516acbf51f01af8480e0f1bd0dd : therefor this commit will be atomic. 

What do you think?

Otherwise I'm good, can be moved to `Test Ready` then.

anansilva commented:
> I would love to see commit https://github.com/openfoodfoundation/openfoodnetwork/commit/ebf53a22f212756d9ac3c41568ba44f8c985a5a7 squashed with commit https://github.com/openfoodfoundation/openfoodnetwork/commit/a1c832a48b5fc516acbf51f01af8480e0f1bd0dd : therefor this commit will be atomic.

Yes, makes sense, squashed. Thanks @jibees!

mkllnk reviewed (approved):
Thank you.

filipefurtad0 commented:
Hey @anansilva ,

I've tested the receipts functionality, before and after your PR.

**before this PR**

Receipts could be activated as superadmin:

![image](https://user-images.githubusercontent.com/49817236/220109952-071a06ab-6d0a-442e-9e07-36726c21395b.png)


The print option was visible, when editing an order:

![image](https://user-images.githubusercontent.com/49817236/220110034-31884f77-0555-4ede-b442-2e0e278d784b.png)


**After this PR**

The receipts option is gone (superadmin):

![image](https://user-images.githubusercontent.com/49817236/220109840-9ba5acb4-f8fa-40c1-8e74-dad3016869d4.png)



The actions menu works just like before, when the receipts option was deactivated:

![image](https://user-images.githubusercontent.com/49817236/220109459-3bdeda89-9d21-484e-a7c4-faa2102ce356.png)

Some regression testing:
- All related pages render as before
- Invoice printing functionality works as before as well

Spotted no regressions, we're good!
And we have ca. less 9400 lines of code :tada: 
![image](https://user-images.githubusercontent.com/49817236/220106954-7ee1f1b2-fa88-42d2-ad59-0a01ba4132a0.png)

Thank you Ana!
Merging.

anansilva commented:
Thanks for the feedback @filipefurtad0!
